Python/pywarpx/_libwarpx.py

```diff
@@ -358,44 +141,21 @@ def _get_boundary_number(self, boundary):
 
         return boundary_num
 
-    @staticmethod
-    def _array1d_from_pointer(pointer, dtype, size):
-        '''
-
-        Function for converting a ctypes pointer to a numpy array
-
-        '''
-        if not pointer:
-            raise Exception(f'_array1d_from_pointer: pointer is a nullptr')
-        if sys.version_info.major >= 3:
-            # from where do I import these? this might only work for CPython...
-            #PyBuf_READ  = 0x100
-            PyBUF_WRITE = 0x200
-            buffer_from_memory = ctypes.pythonapi.PyMemoryView_FromMemory
-            buffer_from_memory.argtypes = (ctypes.c_void_p, ctypes.c_int, ctypes.c_int)
-            buffer_from_memory.restype = ctypes.py_object
-            buf = buffer_from_memory(pointer, dtype.itemsize*size, PyBUF_WRITE)
-        else:
-            buffer_from_memory = ctypes.pythonapi.PyBuffer_FromReadWriteMemory
-            buffer_from_memory.restype = ctypes.py_object
-            buf = buffer_from_memory(pointer, dtype.itemsize*size)
-        return np.frombuffer(buf, dtype=dtype, count=size)
-
     def getNProcs(self):
         '''
 
         Get the number of processors
 
         '''
-        return self.libwarpx_so.warpx_getNProcs()
+        return self.libwarpx_so.getNProcs()
```

Comment:
@ax3l I wrapped the function `amrex::ParallelDescriptor::NProcs` in the `pybind11` module for WarpX (defined in `pyWarpX.cpp`. However, could we instead call these functions directly from `pyamrex`?

Reply:
To avoid code duplication, I think it would be better to rely on the function from pyamrex. And the same for `MyProc` - this should be added to pyamrex.

Reply:
I went ahead and made a PR in pyamrex to add `MyProc` and some other routines. https://github.com/AMReX-Codes/pyamrex/pull/146